projects/skia/build.sh

```diff
@@ -15,91 +15,70 @@
 #
 ################################################################################
 
+# Build SwiftShader
+cd SwiftShader
+export SWIFTSHADER_INCLUDE_PATH=$PWD/include
+rm -rf build
+mkdir build
+
+cd build
+if [ $SANITIZER == "profile" ]; then
+  cmake ..
+else
+  if [ $SANITIZER == "address" ]; then
+    CMAKE_SANITIZER="ASAN"
+  elif [ $SANITIZER == "memory" ]; then
+    CMAKE_SANITIZER="MSAN"
+  elif [ $SANITIZER == "undefined" ]; then
+    CMAKE_SANITIZER="UBSAN"
+  else
+    exit 1
+  fi
+  cmake .. -D$CMAKE_SANITIZER=1
+fi
+
+make -j
+cp libGLESv2.so libEGL.so $OUT
+export SWIFTSHADER_LIB_PATH=$OUT
+
+cd ../../skia
 # These are any clang warnings we need to silence.
 DISABLE="-Wno-zero-as-null-pointer-constant -Wno-unused-template
          -Wno-cast-qual -Wno-self-assign -Wno-return-std-move-in-c++11"
 # Disable UBSan vptr since target built with -fno-rtti.
-export CFLAGS="$CFLAGS $DISABLE -fno-sanitize=vptr"
-export CXXFLAGS="$CXXFLAGS $DISABLE -fno-sanitize=vptr"
+export CFLAGS="$CFLAGS $DISABLE -I$SWIFTSHADER_INCLUDE_PATH -DGR_EGL_TRY_GLES3_THEN_GLES2 -fno-sanitize=vptr"
+export CXXFLAGS="$CXXFLAGS $DISABLE -I$SWIFTSHADER_INCLUDE_PATH -DGR_EGL_TRY_GLES3_THEN_GLES2 -fno-sanitize=vptr"
+export LDFLAGS="-lFuzzingEngine $CXXFLAGS -L$SWIFTSHADER_LIB_PATH"
 
 # This splits a space separated list into a quoted, comma separated list for gn.
 export CFLAGS_ARR=`echo $CFLAGS | sed -e "s/\s/\",\"/g"`
 export CXXFLAGS_ARR=`echo $CXXFLAGS | sed -e "s/\s/\",\"/g"`
-
+export LDFLAGS_ARR=`echo $LDFLAGS | sed -e "s/\s/\",\"/g"`
 
 # Even though GPU is "enabled" for all these builds, none really
 # uses the gpu except for api_mock_gpu_canvas
-$SRC/depot_tools/gn gen out/Fuzz_mem_constraints\
-    --args='cc="'$CC'"
-    cxx="'$CXX'"
-    is_debug=false
-    extra_cflags_c=["'"$CFLAGS_ARR"'"]
-    extra_cflags_cc=["'"$CXXFLAGS_ARR"'","-DIS_FUZZING","-DIS_FUZZING_WITH_LIBFUZZER"]
```

Comment:
The oom/timeout protection code is all wrapped behind 
#ifdef IS_FUZZING, so the only difference between the two builds is the -DIS_FUZZING flag here.  I put them in different folders because it seemed like a good organization thing.

Reply:
Oh right.
I'll bring this back.